diff --git a/src/laserMapping.cpp b/src/laserMapping.cpp
--- a/src/laserMapping.cpp
+++ b/src/laserMapping.cpp
@@ -388,6 +388,11 @@ bool sync_packages(MeasureGroup &meas)
     /*** push a lidar scan ***/
     if(!lidar_pushed)
     {
+        while (lidar_buffer.size() > 1)
+        {
+            lidar_buffer.pop_front();
+            time_buffer.pop_front();
+        }
         meas.lidar = lidar_buffer.front();
         meas.lidar_beg_time = time_buffer.front();
         if (meas.lidar->points.size() <= 1) // time too little
